README.md

```diff
@@ -10,22 +10,105 @@
 [![Chat](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/ManageIQ/manageiq-providers-redfish?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 [![Translate](https://img.shields.io/badge/translate-zanata-blue.svg)](https://translate.zanata.org/zanata/project/view/manageiq-providers-redfish)
 
-ManageIQ plugin for the Redfish provider.
 
-## Development
+## Quickstart
 
-See the section on pluggable providers in the [ManageIQ Developer Setup](http://manageiq.org/docs/guides/developer_setup)
+Redfish provider is installed by default when seting up ManageIQ development
+environment. To start using it, navigate to the `Compute` ->
+`Physical Infrastructure` -> `Providers` and add new provider of type Redfish.
+
+
+## Development environment setup
+
+See the [ManageIQ Developer Setup guide][dev-setup] for help on installing
+prerequistes for ManageIQ development. In this section, we will assume that
+ManageIQ core repo resides in `~/miq/manageiq`, so adjust path accordingly.
+
+   [dev-setup]: https://manageiq.org/docs/guides/developer_setup
+                (ManageIQ developer setup)
+
+Now, to start working Redfish provider, we must first clone the repo:
+
+    $ cd ~/miq
+    $ git clone git@github.com:ManageIQ/manageiq-providers-redfish.git
+    $ cd manageiq-providers-redfish
+
+After this is done, we must perform the initial setup by running
+
+    $ bin/setup
+
+After the initial setup is done, we can test the environment by executing
+
+    $ bundle exec rake
+
+This command should run test suite and return with no error. If this is not
+the case, congratulations to us, since we can start debugging our setup;) If
+everything is green, we are ready start adding ~~bugs~~ features to the
+Redfish provider.
+
+To be able to view and test the changes we are making to the provider, we must
+override the gem sources in `~/miq/manageiq/bundler.d/overrides.rb`. In our
+case, `overrides.rb` should look something like this:
+
+    override_gem "manageiq-providers-redfish",
+      :path => File.expand_path("~/miq/manageiq-providers-redfish")
+
+Do not use relative path in `File.expand_path` call, since this is just
+calling for troubles. To actually install the local gems, run
+
+    $ ( cd ../manageiq && bin/update )
+
+Now, after starting the ManageIQ, the Redfish provider will be gem will be
+installed from local folder instead of being fetched from GitHub.
+
+But it will not take long for us to get into situation where we will need to
+modify some other parts of the ManageIQ in order to get new feature in. In
+order be able to run automated test suite in such situations, we need to
+remove `~/miq/manageiq-providers-redfish/spec/manageiq` folder and replace it
+with symbolic link to `~/miq/manageiq`. To get things updated, we need to run
```

Comment:
Maybe recommend doing this symlink before running `bin/setup` the first time?
`cd ~/miq/manageiq-providers-redfish/spec && ln -s ~/src/manageiq` then `bin/setup`

Reply:
I kept this separate because doing things in one step can feel a bit overwhelming for someone who is only starting to develop MIQ provider (and I would know, since I was there a few months ago). ManageIQ veterans might find this doc a bit verbose because of that, but they are not the primary target of this document.

Reply:
Actually, [plugin development guide](https://github.com/ManageIQ/guides/blob/master/developer_setup/plugins.md) explains this in exactly the same order as I do.

The way I see things (as a newcommer to the ManageIQ), there are three steps to getting things set up:

 1. Clone the provider repo, run `bin/setup`, run `bundle exec rake` and have all tests green: this this step validates that the most of the developer tools are in place and working.
 2. Add override to core in order to test the changes in provider: this step brings in the UI part and ability to interactivelly test the code changes.
 3. Replace `spec/manageiq` subfolder in provider repo with symlink to core repo: this is needed when the user needs to develop core and provider in sync.

The further down the list we go, the more "state" there is for user to manage (keeping repos in sync). And when you starting, first step is already challenging enough.